src/setup_node_env/exit_on_warning.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+if (process.noProcessWarnings !== true) {
+  var ignore = ['MaxListenersExceededWarning'];
+
+  process.on('warning', function(warn) {
```

Comment:
If I'm understanding correctly, we were previously crashing on any deprecation warnings, but now, we will crash on all warnings except `MaxListenersExceededWarning` (including any custom warnings emitted by dependencies or our own code). Is this intended?

Reply:
Previously we used the flags `--trace-warnings --throw-deprecation` when you started Kibana via `yarn start` and we used `--throw-deprecation` in CI. This meant that we would only crash on deprecation warnings and log a stack trace for all other types of warnings (`MaxListenersExceededWarning` belongs to the latter group).

There were a couple of drawbacks to this approach:
1. If the deprecated API was called in a place enclosed in a try-catch (or inside of a Promise or an async/await context), the throw would be caught and the program would not crash.
1. If you ran `./scripts/kibana` directly instead of running `yarn start`, you would not get these flags automatically.
1. If you ran any of our tests locally you would not get the standard CI flags. This meant something that might seem to pass locally would not pass in CI.

This PR changes this behavior by ensuring:
- Crash for all types of warnings - not only deprecation warnings (except `MaxListenersExceededWarning`, though that's just because I didn't want to spend too much time getting that to work).
- That we always crash - no matter if the offending code is surrounded by a try-catch (etc).
- That you always get the same behavior whether you run `yarn start` or `./scripts/kibana`.
- That you always get the same behavior in CI or if you run individual tests locally.

Reply:
Thanks for explaining, that makes sense to me now